syntax = "proto3";






// 查询玩家账号数据 
message ReqDbLoginAuth
{
    string account=1; // 账号
    string password=2; // 密码
    string pf=3; // 平台名字
    sint32 fromServerId=4; // 初始服务器id
    uint64 csessionId=5;
    uint64 sessionId=6;    
}


// 查询玩家账号数据返回 
message ResDbLoginAuth
{
    string account=1; // 账号
    sint32 gmlv=2; // gm等级
    string pf=3; // 平台名字
    sint32 fcmTime=4; //防沉迷已在线时间   
    sint32 code=5; 
    sint32 fromServerId=6; // 初始服务器id
    uint64 csessionId=7;
    uint64 sessionId=8;      
}

// 查询玩家 
message ReqDbSelectPlayer
{
    uint64 csessionId=1; 
    sint32 fromServerId=2;
    string account=3;
    string pf=4;
    uint64 sessionId=5;      
}

// 查询玩家返回 
message ResDbSelectPlayer
{   
    uint64 pid=1; // 玩家id
    uint64 csessionId=2; 
    sint32 code=3;
    uint64 sessionId=4;      
}


// 创建玩家 
message ReqDbCreatePlayer
{
    sint32 sex=1;
    bytes name=2;
    uint64 pid=3; 
    uint64 csessionId=4; 
    string account=5;
    string pf=6;
    sint32 fromServerId=7;
    sint32 serverId=8;
    uint64 sessionId=9;      

}

// 创建玩家返回 
message ResDbCreatePlayer
{
    sint32 sex=1; // 1:男 2:女
    bytes name=2; // 玩家名字
    uint64 pid=3;
    sint32 code=4;
    uint64 csessionId=5; 
    uint64 sessionId=7;      
}


// 请求进入游戏 
message ReqDbEnterGame
{
    uint64 pid=1;  
    uint64 csessionId=2;
    uint64 sessionId=3;         
}

// 请求进入游戏返回 
message ResDbEnterGame
{
    uint64 pid=1; 
    sint32 code=2;   
    uint64 csessionId=3;
    uint64 sessionId=4;       
}

message PlayerBaseData
{
    uint64 pid=1;
    bytes name=2;
    string icon=3;
    string account=4;
    string pf=5;
    sint32 createTime=6;
    uint64 power=7;
    sint32 level=8;
    sint32 vip=9;
    sint32 exp=10;
    uint64 sessionId=11;
    sint32 loginTime=12;
    uint64 guildId=13;
    sint32 chargeVal=14;
    sint32 headIcon=15;
    sint32 title=16;
    sint32 skin=17;
    bytes extra = 18;
    sint32 fromServerId=19;
    sint32 logoutTime=20;
}

message BagData
{
    uint64 guid=1; // 物品唯一id
    uint32 itemId=2; // 物品id
    uint64 count=3; // 数量
    sint32 time=4; // 过期结束时间
    sint32 level=5; // 等级
    sint32 exp=6; // 经验
    uint64 owner=7; // 所有者
    sint32 star=8; // 星级
    sint32 step=9; // 阶级
}

// db返回玩家基础数据 
message ResReturnPlayerBaseData
{
    uint64 pid=1;
    PlayerBaseData data=2;
    uint64 csessionId=3; 
    uint64 sessionId=4;     
}


// db返回玩家背包数据 
message ResReturnPlayerBagData
{
    uint64 pid=1;
    repeated BagData data=2;
    uint64 csessionId=3;
    uint64 sessionId=4;     
}

// db返回玩家模块数据 
message ResReturnPlayerModuleData
{
    sint32 moduleId=1;
    bytes data=2;
    uint64 csessionId=3;
    uint64 sessionId=4;     
}


// db保存玩家基础数据 
message ReqSavePlayerBaseData
{
    uint64 pid=1;
    PlayerBaseData data=2;
}

// 保存玩家背包数据 
message ReqSavePlayerBagData
{
    uint64 pid=1;
    repeated BagData data=2;
}

// db保存玩家模块数据 
message ReqSavePlayerModuleData
{
    uint64 pid=1;
    sint32 moduleId=2;
    bytes data=3;
    sint32 serverId=4;
}

// db发送gm指令到game 
message ReqSendGmToGame
{
    string cmd=1;
}

// game进程上报自身数据
message ReqGameReport
{
    sint32 serverId=1;
}


message ReqGameQuit
{

}

message LogItems
{
    sint32 id=1;
    uint64 cnt=2;
    uint64 oldCnt=3;
}

// 写道具日志 
message WriteLogData
{
    uint64 pid=1;
    string account=2;
    string pf=3;
    bytes desc=4;
    sint32 serverId=5;
    bytes extra=6;
    bytes name=7;
    repeated LogItems data=8;
}


message LogMails
{
    uint64 mailId=1;
    bytes title=2;
    bytes content=3;  
    string reward=4;
    bytes desc=5;// 邮件描述
    sint32 expireTime=6;   
    bytes extra=7; 
}

// 写邮件日志 
message WriteMailData
{
    uint64 pid=1;
    string account=2;
    string pf=3;
    sint32 serverId=4;
    bytes name=5;
    repeated LogMails data=6;
}